org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/SaveActionExtensions.scala

```diff
@@ -153,31 +147,63 @@ trait SaveActionExtensions extends HasLogger {
   }
 
   /**
-   * Applies all save actions that extends [[DocumentSupport]].
+   * Applies all save actions that extend [[DocumentSupport]].
    */
   private def applyDocumentExtensions(udoc: IDocument): Unit = {
     for ((setting, ext) <- documentSaveActions if isEnabled(setting.id)) {
       val doc = new TextDocument(udoc)
       val instance = ext(doc)
-      performExtension(instance, udoc) {
+      performDocumentExtension(instance, udoc) {
         instance.perform()
       }
     }
   }
 
   /**
-   * Applies all save actions that extends [[CompilerSupport]].
+   * Applies all save actions that extend [[CompilerSupport]].
    */
   private def applyCompilerExtensions(udoc: IDocument): Unit = {
-    for ((setting, ext) <- compilerSaveActions if isEnabled(setting.id)) {
-      createExtensionWithCompilerSupport(ext) foreach { instance =>
-        performExtension(instance, udoc) {
-          instance.global.asInstanceOf[IScalaPresentationCompiler].asyncExec {
-            instance.perform()
-          }.getOrElse(Seq())()
+    val timeout = saveActionTimeout
+
+    def loop(xs: Seq[(SaveActionSetting, CompilerSupportCreator)]): Unit = xs match {
+      case Nil ⇒
+
+      case (setting, ext) :: xs if isEnabled(setting.id) ⇒
```

Comment:
`xs` is a `Seq`, but `::` matches only if underlying implementation is a `List`. Using `+:` is more safe here.